swarm/network/stream/delivery.go

```diff
@@ -178,22 +94,15 @@ func (d *Delivery) handleRetrieveRequestMsg(ctx context.Context, sp *Peer, req *
 			log.Debug("ChunkStore.Get can not retrieve chunk", "peer", sp.ID().String(), "addr", req.Addr, "hopcount", req.HopCount, "err", err)
 			return
 		}
-		if req.SkipCheck {
-			syncing := false
-			osp.LogFields(olog.Bool("skipCheck", true))
+		syncing := false
+		osp.LogFields(olog.Bool("skipCheck", true))
 
-			err = sp.Deliver(ctx, ch, s.priority, syncing)
-			if err != nil {
-				log.Warn("ERROR in handleRetrieveRequestMsg", "err", err)
-			}
-			osp.LogFields(olog.Bool("delivered", true))
-			return
-		}
-		osp.LogFields(olog.Bool("skipCheck", false))
-		select {
-		case streamer.deliveryC <- ch.Address()[:]:
-		case <-streamer.quit:
+		err = sp.Deliver(ctx, ch, 1, syncing)
```

Comment:
It is be better to use named priority as the third argument.

Reply:
You're right. Priority queue will be removed in a follow up PR, but since it is not here, I am just using `Top`. Let me know if this priority makes sense.

To me the priority queue is redundant and we don't have any proof that it is actually needed and not premature optimization. I've asked multiple times about this, and have never received a straight answer about it.

Reply:
I think that Mid priority is 1 and Top is 3. If you plan to remove the priority queue, then it does not matter so much.